### PR TITLE
refactor(statesync): implement statesync peer manager

### PR DESCRIPTION
**Pull request by shotonoff**
## Issue being fixed or feature implemented

This is PR is related to the task TD-162:

Each module that communicates with p2p peers has its own simple peer manager implementation. Usually the code is placed into a reactor. As a result of the refactoring, peer management has been moved out of the reactor and into a separate component.
The component must listen to peer `up` and `down` events and register the peer into the peer store or remove it.

## What was done?

* Migrate p2p `statesync` channel descriptors in p2p module
* Declare `SnapshotClient` interface and implement it
* Implement peer-manager component and peer-update subscriber 
* Additionally create `LightBlockRepository`

## How Has This Been Tested?

Add a couple unit-tests

## Breaking Changes

N/A

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: internal/statesync/repo.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		n = recentSnapshots
+	}
+	snapshots := make([]*snapshot, 0, n)
+	for _, s := range resp.Snapshots[:n] {
+		snapshots = append(snapshots, newSnapshotFromABCI(s))
+	}
+	return snapshots, nil
+}
+
+func sortSnapshot(snapshots []*abci.Snapshot) {
```
Comment (lklimek) [inline internal/statesync/repo.go:138]:
```suggestion
func sortSnapshots(snapshots []*abci.Snapshot) {
```

File: internal/p2p/channel_params.go [hunk 2]
```diff
@@ -75,15 +136,16 @@ func ResolveChannelID(msg proto.Message) ChannelID {
...
-		*statesync.SnapshotsRequest,
-		*statesync.SnapshotsResponse,
-		*statesync.ChunkRequest,
-		*statesync.ChunkResponse,
-		*statesync.LightBlockRequest,
-		*statesync.LightBlockResponse,
-		*statesync.ParamsRequest,
-		*statesync.ParamsResponse:
+		*consensus.HasCommit:
+		// TODO: enable these channels when they are implemented
```
[the remark this answers is not in the document]
Comment (shotonoff) [reply internal/p2p/channel_params.go:140]: This TODO should be deleted and lines below uncommented once we filly switch on a new approach

File: internal/peer/store.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+import (
+	sync "github.com/sasha-s/go-deadlock"
+
+	"github.com/tendermint/tendermint/types"
+)
+
+type (
+	Store[T any] interface {
+		Get(peerID types.NodeID) (T, bool)
```
[the remark this answers is not in the document]
Comment (shotonoff) [reply internal/peer/store.go]: Add a generic type for a key, and move this component into `libs/store` package to make this sharable

File: internal/statesync/peer.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+type (
+	// HandlerFunc is peer update event handler
+	HandlerFunc func(ctx context.Context, update p2p.PeerUpdate) error
+	// PeerManager is a manager for peers
+	PeerManager struct {
+		logger    log.Logger
+		client    client.SnapshotClient
+		peerStore peer.Store[PeerData]
+		peerSubs  *PeerSubscriber
+		stopCh    chan struct{}
```
[the remark this answers is not in the document]
Comment (shotonoff) [reply internal/statesync/peer.go]: fixed

**Review by shotonoff — COMMENTED** [review]
PR is ready to new round review

**Review by lklimek — DISMISSED** [review]

**Review by lklimek — APPROVED** [review]

**Pull request merged**